src/EditorFeatures/Test/EmbeddedLanguages/StackFrame/StackFrameSyntaxFactory.cs

```diff
@@ -0,0 +1,204 @@
+        public static readonly StackFrameToken EOLToken = CreateToken(StackFrameKind.EndOfLine, "");
+        public static readonly StackFrameToken ColonToken = CreateToken(StackFrameKind.ColonToken, ":");
+
+        public static readonly StackFrameTrivia AtTrivia = CreateTrivia(StackFrameKind.AtTrivia, "at ");
+        public static readonly StackFrameTrivia LineTrivia = CreateTrivia(StackFrameKind.LineTrivia, "line ");
+        public static readonly StackFrameTrivia InTrivia = CreateTrivia(StackFrameKind.InTrivia, " in ");
+
+        public static readonly StackFrameParameterList EmptyParams = ParameterList(OpenParenToken, CloseParenToken);
+
+        public static StackFrameParameterDeclarationNode Parameter(StackFrameNameNode type, StackFrameToken identifier)
+            => new(type, identifier);
+
+        public static StackFrameParameterList ParameterList(params StackFrameParameterDeclarationNode[] parameters)
+            => ParameterList(OpenParenToken, CloseParenToken, parameters);
+
+        public static StackFrameParameterList ParameterList(StackFrameToken openToken, StackFrameToken closeToken, params StackFrameParameterDeclarationNode[] parameters)
+        {
+            var separatedList = parameters.Length == 0
+                ? SeparatedStackFrameNodeList<StackFrameParameterDeclarationNode>.Empty
+                : new(CommaSeparateList(parameters));
+
+            return new(openToken, separatedList, closeToken);
+
+            static ImmutableArray<StackFrameNodeOrToken> CommaSeparateList(StackFrameParameterDeclarationNode[] parameters)
+            {
+                var builder = ImmutableArray.CreateBuilder<StackFrameNodeOrToken>();
+                builder.Add(parameters[0]);
+
+                for (var i = 1; i < parameters.Length; i++)
+                {
+                    builder.Add(CommaToken);
+                    builder.Add(parameters[i]);
+                }
+
+                return builder.ToImmutable();
+            }
+        }
+
+        public static StackFrameMethodDeclarationNode MethodDeclaration(
+            StackFrameQualifiedNameNode memberAccessExpression,
+            StackFrameTypeArgumentList? typeArguments = null,
+            StackFrameParameterList? argumentList = null)
+        {
+            return new StackFrameMethodDeclarationNode(memberAccessExpression, typeArguments, argumentList ?? ParameterList(OpenParenToken, CloseParenToken));
+        }
+
+        public static StackFrameQualifiedNameNode QualifiedName(string s, StackFrameTrivia? leadingTrivia = null, StackFrameTrivia? trailingTrivia = null)
+            => QualifiedName(s, CreateTriviaArray(leadingTrivia), CreateTriviaArray(trailingTrivia));
+
+        public static StackFrameQualifiedNameNode QualifiedName(string s, ImmutableArray<StackFrameTrivia> leadingTrivia, ImmutableArray<StackFrameTrivia> trailingTrivia)
+        {
+            StackFrameNameNode? current = null;
```

Comment:
Add assert that 's' contains a dot.  otherwise, the cast to StackFrameQualifiedNameNode below seems suspect.

Reply:
Sure